crates/oxc_parser/src/js/function.rs

```diff
@@ -74,6 +74,18 @@ impl<'a> Parser<'a> {
         Ok(self.ast.formal_parameters(self.end_span(span), params_kind, list.elements, list.rest))
     }
 
+    pub(crate) fn parse_formal_parameters_with_ts_this(
+        &mut self,
+        params_kind: FormalParameterKind,
+    ) -> Result<(Option<TSThisParameter<'a>>, Box<'a, FormalParameters<'a>>)> {
+        let span = self.start_span();
```

Comment:
I think we need a "if source is typescript" check here?

Reply:
If the source code is not TypeScript, then `this_param` will always be `None`, and the function will have the exact same effect as `parse_formal_parameters`.

The purpose of this function is to extract and separate `this_param` from the list and return it. This function's name may be misleading, perhaps it can be changed to a more suitable one?